numalogic/config/factory.py

```diff
@@ -24,24 +26,26 @@
 from numalogic.models.threshold import StdDevThreshold, StaticThreshold, SigmoidThreshold
 from numalogic.postprocess import TanhNorm, ExpMovingAverage
 from numalogic.preprocess import LogTransformer, StaticPowerTransformer, TanhScaler
+from numalogic.registry import MLflowRegistry
+from numalogic.registry.redis_registry import RedisRegistry
 from numalogic.tools.exceptions import UnknownConfigArgsError
 
 
 class _ObjectFactory:
     _CLS_MAP = {}
 
-    def get_instance(self, model_info: ModelInfo):
+    def get_instance(self, object_info: Union[ModelInfo, RegistryInfo]):
         try:
-            _cls = self._CLS_MAP[model_info.name]
+            _cls = self._CLS_MAP[object_info.name]
         except KeyError:
-            raise UnknownConfigArgsError(f"Invalid model info instance provided: {model_info}")
-        return _cls(**model_info.conf)
+            raise UnknownConfigArgsError(f"Invalid model info instance provided: {object_info}")
```

Comment:
Another design pattern issue: exception should only be used in exceptional cases. Here, we should simply check if object_info.name in _CLS_MAP.

That being said, this comment may be moot if we do dynamic import to reduce the footprint of the library.

Reply:
Following this:
(https://docs.python.org/3.6/glossary.html#term-eafp)
Easier to ask for forgiveness than permission. This common Python coding style assumes the existence of valid keys or attributes and catches exceptions if the assumption proves false. This clean and fast style is characterized by the presence of many [try](https://docs.python.org/3.6/reference/compound_stmts.html#try) and [except](https://docs.python.org/3.6/reference/compound_stmts.html#except) statements.